source/Halibut/Transport/Protocol/MessageExchangeStream.cs

```diff
@@ -235,9 +235,9 @@ void ExpectServerIdentity()
 
         T ReadBsonMessage<T>()
         {
-            using (var buffer = new BufferedStream(stream, 8192, true))
-            using (var zip = new DeflateStream(buffer, CompressionMode.Decompress, true))
-            using (var bson = new BsonDataReader(zip) { CloseInput = false })
+            using (var zip = new DeflateStream(stream, CompressionMode.Decompress, true))
+            using (var buffer = new BufferedStream(new NonDisposableStreamWrapper(zip), 8192))
```

Comment:
The blog post only mentions `Compress` wouldn't `Decompress` use the reverse logic?

I would be good to get some metrics around this change to confirm it works as intended.

Reply:
The read part of the code is interesting because it makes me wonder if we even need to buffer at all, to me ` return (T)serializer.Deserialize<MessageEnvelope>(bson).Message;` needs the whole payload anyway.

Reply:
@droyad any thoughts here?

Reply:
Yeah this feels like a copy-n-paste error from the write part. We don't need buffering on read.

Reply:
After dealing with this on the Server Nancy side, I think it's best to run a test on this. Perhaps use the console apps to test out the transfer speed of a large file from the US to AU with the various combinations. 

This might actually be the cause of the slowness some customers are reporting when transferring packages.